lib/easypost/shipment.rb

```diff
@@ -183,4 +183,14 @@ def generate_form(form_type, form_options = {})
 
     self
   end
+
+  # Retrieves the estimated delivery date of each Rate via SmartRate.
+  def retrieve_estimated_delivery_date(planned_ship_date)
+    url = "#{self.url}/smartrate/delivery_date"
+    params = {}
+    params[:planned_ship_date] = planned_ship_date
```

Comment:
suggest: these should be consolidated. When we know all the params we shouldn't need to build an empty object just to then add to it, create the object and populate it in one step

Reply:
Was following the above code, consolidated!